### PR TITLE
applications: serial_lte_modem: at_host: use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by gmarull**
Obtain UART devices at compile time using DEVICE_DT_GET.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

### CODE REVIEW

File: applications/serial_lte_modem/src/slm_at_host.c [hunk 1]
```diff
@@ -956,16 +956,16 @@ int slm_at_host_init(void)
 
 	/* Initialize the UART module */
 #if defined(CONFIG_SLM_CONNECT_UART_0)
-	uart_dev = device_get_binding(DT_LABEL(DT_NODELABEL(uart0)));
+	uart_dev = DEVICE_DT_GET(DT_NODELABEL(uart0));
```
Comment (anangl) [inline applications/serial_lte_modem/src/slm_at_host.c:959]: Why not initialize it at build time? `uart_dev` could be `const` then.

**Review by pirun — APPROVED** [review]

File: applications/serial_lte_modem/src/slm_at_host.c (hunk 1 above)
Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_at_host.c:959]: Can share piece of sample code?

Comment (anangl) [reply applications/serial_lte_modem/src/slm_at_host.c:959]: I mean something like:
```
#if defined(CONFIG_SLM_CONNECT_UART_0)
#define UART_NODE DT_NODELABEL(uart0)
#elif defined(CONFIG_SLM_CONNECT_UART_2)
#define UART_NODE DT_NODELABEL(uart2)
#else
#error "Unsupported UART instance"
#endif
static const struct device *const uart_dev = DEVICE_DT_GET(UART_NODE);
```

Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_at_host.c:959]: Looks nice, thanks!
Please add by this PR.

**Review by junqingzou — APPROVED** [review]

File: applications/serial_lte_modem/src/slm_at_host.c [hunk 2]
```diff
@@ -49,7 +57,7 @@ static enum slm_operation_modes {
 	SLM_DFU_MODE          /* nRF52 DFU controller */
 } slm_operation_mode;
 
-static const struct device *uart_dev;
+static const struct device *uart_dev = DEVICE_DT_GET(UART_NODE);
```
Comment (anangl) [inline applications/serial_lte_modem/src/slm_at_host.c:60]:
```suggestion
static const struct device *const uart_dev = DEVICE_DT_GET(UART_NODE);
```

**Review by anangl — APPROVED** [review]

**Pull request merged**